src/node_modules/_file-explorer.js

```diff
@@ -0,0 +1,292 @@
+var setval = require('setval')
+var yo = require('yo-yo')
+var csjs = require('csjs-inject')
+var Treeview = require('ethereum-remix').ui.TreeView
+
+var css = csjs`
+  .treeview {
+    display: block;
+    margin-top: 7px;
+    width: 20%;
+  }
+  .folder,
+  .file {
+    font-size: 14px;
+  }
+  .hasFocus {
+    background-color: #F4F6FF;
+  }
+  .rename {
+    background-color: white;
+  }
+  .remove,
+  .save {
+    align-self: center;
+    margin-right: 3%;
+  }
+  .activeMode {
+    display: flex;
+    justify-content: space-between;
+  }
+`
+module.exports = fileExplorer
+
+function fileExplorer (container, files, editor) {
```

Comment:
I can highly recommend a manually managed `src/node_modules/` folder which even allows to specify the relevant transforms if necessary. It really is awesome in making sure modules dont know anything about each other.

Otherwise - the file-explorer module the way it's currently written works a bit the other way around. It treats `files` as some kind of database singleton that might or might not sync with a backend and is passed everywhere.

I did that, because it was already very very hard to have any kind of overview in the `app.js` file, because basically everything is going on in the `app.js` file with no way of knowing - other than by convention - what is accessing which other things.

`app.js` feels like a very polluted global space.

The `files` object and accessing it's api is so essential to file-explorer, that i feel it basically means copy/pasting it into `app.js`. ...not sure what to do about it... but maybe, what about, for example:

```js
// inside file-explorer component
function removeSubtree (files, path) {
  var allPaths = Object.keys(files.list())
  var remove = allPaths.filter(function keep (p) { return ~p.indexOf(path) })
  remove.forEach(function (p) { api.files.emit('remove', p) }) // instead of `files.remove(p)`
}
// or
var filenames
api.files.emit('list', function (list) {
  filenames = list
}) // instead of `var filenames = files.list()`
```
and then in `app.js`
```js
var api = fileExplorer() // or `new FileExplorer()`
api.on('list', function (cb) { cb(files.list()) })
api.on('remove', function (path, cb) { cb(files.remove(p)) })
```
... and so on.

would that work?